### PR TITLE
Fixes #9194 - add ability to copy fact value to clipboard

### PR DESCRIPTION
**Pull request by shlomizadok**
Add ability to copy a fact value to clipboard.
Sadly, our bootstrap version does not support "copy" or "copy to clipboard" glyphicons so I've used the "paperclip" icon.

### CODE REVIEW

**Comment by orrabin** [comment]
@shlomizadok maybe add the `content_tag` to the `trunc_with_tooltip` helper?
I think this could be useful in other places that trunc values too

**Comment by shlomizadok** [comment]
I tend to disagree as many of the `trunc_with_tooltip` are wrapped in a link - and I'd like to avoid adding the clipboard into the link

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -110,6 +111,8 @@ function onContentLoad(){
 
   var tz = jstz.determine();
   $.cookie('timezone', tz.name(), { path: '/' });
+
+  var clip = new ZeroClipboard($(".fact_clip_button"))
```
Comment (ohadlevy) [inline app/assets/javascripts/application.js]: can we generalize the name? i assume we might want to copy from other fields in general (e.g. parameter) and fact_clip is not great as its in app.js

File: app/assets/stylesheets/bootstrap_and_overrides.scss [hunk 2]
```diff
@@ -194,3 +194,8 @@ span.btn a{ text-decoration: none; color: #333333}
 body.modal-open {
   overflow: visible;
 }
+
+
+.glyphicon-copy:before {
+  content: "\f121";
```
Comment (ohadlevy) [inline app/assets/stylesheets/bootstrap_and_overrides.scss]: whats the background here? if its a workaround, from which version should we remove it?

Comment (shlomizadok) [inline app/assets/stylesheets/bootstrap_and_overrides.scss]: This was an attempt to use copy glyphicon, which is not supported in version 3.0.3 (our current version) - yet it didn't work. Removing this as eventually I've used paperclip glyphicon.

File: bundler.d/assets.rb [hunk 3]
```diff
@@ -17,4 +17,5 @@
   gem 'gridster-rails', '~> 0.1'
   gem 'jquery_pwstrength_bootstrap', '~> 1.2'
   gem 'jquery-turbolinks', '~> 2.1'
+  gem 'zeroclipboard-rails', '~> 0.1'
```
Comment (ohadlevy) [inline bundler.d/assets.rb]: warning flash hidden here :boom:

Comment (shlomizadok) [inline bundler.d/assets.rb]: :octocat: uses it too. Also hiding the button if flash is not installed. but basically yeah :boom:

**Comment by dLobatog** [comment]
Sorry, I'm a bit confused, why do we need to copy the fact value in this way? If we trunc the value what @orrabin should be enough and introduce 0 complexity in my opinion, and keeps us from adding another assets dependency.

**Comment by shlomizadok** [comment]
Have you tried copying long values? like the value of `sshecdsakey` ? the value is at least twice longer than the truncated value and it is impossible to copy the full value.

**Comment by domcleal** [comment]
> Have you tried copying long values? like the value of sshecdsakey ? the value is at least twice longer than the truncated value and it is impossible to copy the full value.

See also: http://projects.theforeman.org/issues/9336

**Comment by dLobatog** [comment]
[test] because of the latest enabled cops

**Comment by shlomizadok** [comment]
Shall I bother rebasing this pr or shall I close it?

**Comment by ohadlevy** [comment]
[test]

**Comment by shlomizadok** [comment]
@domcleal @eLobato  - any thoughts on this PR? If the objection to flash is to big, I can close this.

**Comment by dLobatog** [comment]
I think what's mentioned in [#9336](http://projects.theforeman.org/issues/9336) is a better approach, it introduces no dependencies too. Something along the lines of a button :heavy_plus_sign:  that displays the whole content when needed, like we do with the password in `app/views/hosts/console/vnc`

**Comment by shlomizadok** [comment]
removed the flash. Added :heavy_plus_sign: to long values (> 40)

**Comment by ohadlevy** [comment]
any alternative icon? the plus pretty large?

**Comment by shlomizadok** [comment]
Made the :heavy_plus_sign: smaller

**Comment by dLobatog** [comment]
@shlomizadok Great, we're almost there. Two minor comments:

It feels a little awkward to look right to expand the value, can you put the plus next to the `...`?


The border doesn't fit very well in the foreman theme, I can see why it's there though. 
How about hiding the link on top, and displaying the content with no border when it's expanded? Substituting one text for the other should make the visual effect of the text actually being expanded.


Sorry for the nitpicks especially after such a long time, thanks for the update, this can be very helpful.

**Comment by shlomizadok** [comment]
@eLobato - changed to as you requested above.

**Comment by dLobatog** [comment]
Merged as d254c79ad1cead49e50dc5cd58646015dd1ab35d, thanks @shlomizadok!

**Pull request closed without merging**